libraries/extractor/src/main/java/androidx/media3/extractor/DtsUtil.java

```diff
@@ -223,36 +352,365 @@ public static int getDtsFrameSize(byte[] data) {
     return uses14BitPerWord ? fsize * 16 / 14 : fsize;
   }
 
-  private static ParsableBitArray getNormalizedFrameHeader(byte[] frameHeader) {
-    if (frameHeader[0] == FIRST_BYTE_BE) {
+  /**
+   * Parses the DTS audio format from the extension substream header of a DTS-HD frame according to
+   * ETSI TS 102 114 V1.6.1 (2019-08), Section 7.4/7.5.
+   *
+   * @param frame The DTS-HD frame containing extension substream header.
+   * @return The DTS audio format extracted from the header.
+   */
+  public static DtsAudioFormat parseDtsHdFormat(byte[] frame) throws ParserException {
+    ParsableBitArray frameBits = getNormalizedFrame(frame);
+    frameBits.skipBits(32 + 8); // SYNCEXTSSH, UserDefinedBits
+
+    int extensionSubstreamIndex = frameBits.readBits(2); // nExtSSIndex
+    int headerBits; // nuBits4Header
+    int extensionSubstreamFrameSizeBits; // nuBits4ExSSFsize
+    if (!frameBits.readBit()) { // bHeaderSizeType
+      headerBits = 8;
+      extensionSubstreamFrameSizeBits = 16;
+    } else {
+      headerBits = 12;
+      extensionSubstreamFrameSizeBits = 20;
+    }
+    frameBits.skipBits(headerBits); // nuExtSSHeaderSize
+    int extensionSubstreamFrameSize =
+        frameBits.readBits(extensionSubstreamFrameSizeBits) + 1; // nuExtSSFsize
+
+    int assetsCount; // nuNumAssets
+    int referenceClockCode; // nuRefClockCode
+    int extensionSubstreamFrameDurationCode; // nuExSSFrameDurationCode
+
+    boolean staticFieldsPresent = frameBits.readBit(); // bStaticFieldsPresent
+    if (staticFieldsPresent) {
+      referenceClockCode = frameBits.readBits(2);
+      extensionSubstreamFrameDurationCode = 512 * (frameBits.readBits(3) + 1);
+
+      if (frameBits.readBit()) { // bTimeStampFlag
```

Comment:
this bit is hard to match against the spec imo

AFAICT the if (frameBits.readBit()) { line here corresponds to m_bParamPresent = (ExtractBits(1)==1) ? TRUE : FALSE; in Table 5-3.

Which then means we're assuming that m_bSyncFramePredefValueExists = false? Because if it's true I think we need to read another bit (m_bUpdateFlag = (m_bSyncFramePredefValueExists) ? (bool) ExtractBits(1) : TRUE;)

Do you know what m_bSyncFramePredefValueExists corresponds to in the code (and thus why it's correct to assume it's true)? Maybe worth a bit more detail in the comment to explain the reasoning here.

Reply:
I have updated the comment. Please refer [b8f0350](https://github.com/androidx/media/pull/275/commits/b8f0350ddbb50c6d74b758c99d40d9d674c61009)

Reply:
The comment still doesn't explain why `m_bSyncFramePredefValueExists` is false?

Reply:
The m_bSyncFramePredefValueExists is set to false in the encoder for the m_TimeStamp field.